src/composer/src/cossystem.rs

```diff
@@ -28,13 +28,20 @@ pub struct InterfaceVariant {
     pub variant: Option<String>,
 }
 
+#[derive(Debug, Deserialize,Clone)]
```

Comment:
Please make sure to run `rustfmt` on all of this. You should never do a commit without running `rustfmt`.